ipc/loader.py

```diff
@@ -0,0 +1,53 @@
+#  Copyright (C) 2017, Jaguar Land Rover
+#
+#  This program is licensed under the terms and conditions of the
+#  Mozilla Public License, version 2.0.  The full text of the 
+#  Mozilla Public License is at https://www.mozilla.org/MPL/2.0/
+#
+
+import inspect
+import importlib
+
+class LoaderError(Exception):
+    """Base exception for all plugin loader errors"""
+    pass
+
+def _load_module(modulename):
+    try:
+        return importlib.import_module(modulename)
+    except ImportError:
+        raise LoaderError("error loading module: {}".format(modulename))
+
+def _method_exists(module, method):
+    return hasattr(module, method) and inspect.isroutine(getattr(module, method))
+
+def load_plugin(modulename, use_receive=True, use_send=True):
+    """
+    This method will load plugin 'modulename'.
+
+    It also offers the two optional arguments:
+
+    use_receive: Load receive function.
+    use_send:    Load send function.
+    """
+    # Load plugin module.
+    module = _load_module(modulename)
+
+    # Inspect caller module.
+    caller_info = inspect.stack()[1]
+    caller_module = inspect.getmodule(caller_info[0])
+
+    # Override receive/send functions.
+    if use_receive:
```

Comment:
I see. Yeah, that is not a use case we care about as the VSM is not going to be an IPC library, but rather a processing engine. So we just need to implement use cases that use it as blackbox tied to other clients that already know how to use the IPC in question (such as zeromq).

I think we should keep it simple. For the test case you mention we should not need to share any code since it is very simple. If we end up doing more complex translation we can factor that complex translation into a module and using it on the test code.